sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestController.java

```diff
@@ -161,120 +162,238 @@ public static void showCaseUpdateWithNewDiseaseVariantDialog(CaseDataDto existin
 					FacadeProvider.getCaseFacade().saveCase(caseDataByUuid);
 					ControllerProvider.getCaseController().navigateToCase(caseDataByUuid.getUuid());
 				}
+				if (callback != null) {
+					callback.run();
+				}
 			});
 	}
 
-	private void savePathogenTest(PathogenTestDto dto, BiConsumer<PathogenTestDto, Runnable> onSavedPathogenTest) {
-		facade.savePathogenTest(dto);
+	public PathogenTestDto savePathogenTest(PathogenTestDto dto, BiConsumer<PathogenTestDto, Runnable> onSavedPathogenTest) {
+		PathogenTestDto savedDto = facade.savePathogenTest(dto);
 		final SampleDto sample = FacadeProvider.getSampleFacade().getSampleByUuid(dto.getSample().getUuid());
 		final CaseReferenceDto associatedCase = sample.getAssociatedCase();
 		final ContactReferenceDto associatedContact = sample.getAssociatedContact();
 		final EventParticipantReferenceDto associatedEventParticipant = sample.getAssociatedEventParticipant();
 		if (associatedCase != null) {
 			handleAssociatedCase(dto, onSavedPathogenTest, associatedCase);
-		} else if (associatedContact != null) {
+		}
+		if (associatedContact != null) {
 			handleAssociatedContact(dto, onSavedPathogenTest, associatedContact);
-		} else if (associatedEventParticipant != null) {
+		}
+		if (associatedEventParticipant != null) {
 			handleAssociatedEventParticipant(dto, onSavedPathogenTest, associatedEventParticipant);
 		}
+		return savedDto;
 	}
 
 	private void handleAssociatedCase(
 		PathogenTestDto dto,
 		BiConsumer<PathogenTestDto, Runnable> onSavedPathogenTest,
 		CaseReferenceDto associatedCase) {
+
+		// Negative test result AND test result verified
+		// a) Tested disease == case disease AND test result != sample pathogen test result: Ask user whether to update the sample pathogen test result
+		// b) Tested disease != case disease: Do nothing
+
+		// Positive test result AND test result verified
+		// a) Tested disease == case disease: Ask user whether to update the sample pathogen test result
+		// a.1) Tested disease variant != case disease variant: Ask user to change the case disease variant
+		// a.2) Case classification != confirmed: Ask user whether to confirm the case
+		// b) Tested disease != case disease: Ask user to create a new case for the tested disease
+
 		CaseDataDto preSaveCaseDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(associatedCase.getUuid());
```

Comment:
This does not make sense. `preSaveCaseDto` is used here to be compared against `postSaveCaseDto` to determine which notification to show.
However, there is no chance of `preSaveCaseDto` being different from (or more specifically: "having a different classification than", see `showSaveNotification`) `postSaveCaseDto`, since both are freshly retrieved from the database.